src/ui/public/vis/vis_update_state.js

```diff
@@ -49,6 +49,13 @@ function convertTermAggregation(visState) {
   }
 }
 
+function convertPropertyNames(visState) {
+  if (typeof _.get(visState, 'params.showMeticsAtAllLevels') === 'boolean') {
```

Comment:
This logic is really clear to me, but if someone does another spell-check I can imagine them benefiting from a comment so they won't need to parse it. What do you think of adding this:

```js
// 'showMeticsAtAllLevels' is a legacy typo we'll fix by changing it to 'showMetricsAtAllLevels'.
```

Reply:
with 7.0 we will be removing this whole function and moving the logic to the migration tool

Reply:
Awesome!